includes/class-wc-stripe-helper.php

```diff
@@ -61,144 +97,142 @@ public static function delete_main_stripe_settings() {
 	 * Gets the Stripe currency for order.
 	 *
 	 * @since 4.1.0
-	 * @param object $order
+	 * @param WC_Stripe_Order $order
 	 * @return string $currency
+	 *
+	 * @deprecated 9.3.0
 	 */
 	public static function get_stripe_currency( $order = null ) {
+		_deprecated_function( __METHOD__, '9.3.0', 'WC_Stripe_Order::get_stripe_currency' );
 		if ( is_null( $order ) ) {
 			return false;
 		}
 
-		return $order->get_meta( self::META_NAME_STRIPE_CURRENCY, true );
+		return $order->get_stripe_currency();
 	}
 
 	/**
 	 * Updates the Stripe currency for order.
 	 *
 	 * @since 4.1.0
-	 * @param object $order
+	 * @param WC_Stripe_Order $order
 	 * @param string $currency
+	 *
+	 * @deprecated 9.3.0
 	 */
 	public static function update_stripe_currency( $order, $currency ) {
+		_deprecated_function( __METHOD__, '9.3.0', 'WC_Stripe_Order::set_stripe_currency' );
 		if ( is_null( $order ) ) {
 			return false;
 		}
 
-		$order->update_meta_data( self::META_NAME_STRIPE_CURRENCY, $currency );
+		$order->set_stripe_currency( $currency );
 	}
 
 	/**
 	 * Gets the Stripe fee for order. With legacy check.
 	 *
 	 * @since 4.1.0
-	 * @param object $order
+	 * @param WC_Stripe_Order $order
 	 * @return string $amount
+	 *
+	 * @deprecated 9.3.0
 	 */
 	public static function get_stripe_fee( $order = null ) {
+		_deprecated_function( __METHOD__, '9.3.0', 'WC_Stripe_Order::get_fee' );
 		if ( is_null( $order ) ) {
 			return false;
 		}
 
-		$amount = $order->get_meta( self::META_NAME_FEE, true );
-
-		// If not found let's check for legacy name.
-		if ( empty( $amount ) ) {
-			$amount = $order->get_meta( self::LEGACY_META_NAME_FEE, true );
-
-			// If found update to new name.
-			if ( $amount ) {
-				self::update_stripe_fee( $order, $amount );
-			}
-		}
-
-		return $amount;
+		return $order->get_fee();
 	}
 
 	/**
 	 * Updates the Stripe fee for order.
 	 *
 	 * @since 4.1.0
-	 * @param object $order
+	 * @param WC_Stripe_Order $order
 	 * @param float  $amount
+	 *
+	 * @deprecated 9.3.0
 	 */
 	public static function update_stripe_fee( $order = null, $amount = 0.0 ) {
+		_deprecated_function( __METHOD__, '9.3.0', 'WC_Stripe_Order::set_fee' );
 		if ( is_null( $order ) ) {
 			return false;
 		}
 
-		$order->update_meta_data( self::META_NAME_FEE, $amount );
+		$order->set_fee( $amount );
 	}
 
 	/**
 	 * Deletes the Stripe fee for order.
 	 *
 	 * @since 4.1.0
-	 * @param object $order
+	 * @param WC_Stripe_Order $order
+	 *
+	 * @deprecated 9.3.0
 	 */
 	public static function delete_stripe_fee( $order = null ) {
+		_deprecated_function( __METHOD__, '9.3.0', 'WC_Stripe_Order::delete_fee' );
 		if ( is_null( $order ) ) {
 			return false;
 		}
 
-		$order->delete_meta_data( self::META_NAME_FEE );
-		$order->delete_meta_data( self::LEGACY_META_NAME_FEE );
+		$order->delete_fee();
```

Comment:
```suggestion
		$order->delete_meta_data( self::META_NAME_FEE );
		$order->delete_meta_data( self::LEGACY_META_NAME_FEE );
```


@wjrosa I am still getting the fatal error when I try to renew a subscription from the subscription edit page. The reason is , though it's calling this helper method now, the helper method itself tries to call the function from the new order wrapper class.

<img width="916" alt="Screenshot 2025-02-27 at 6 05 04 PM" src="https://github.com/user-attachments/assets/8c5159fa-1c5f-4b1b-b667-0aa3a6b3ba8d" />

This helps to realize another problem, if any third party plugins are using these helper methods, they would send an instance of `WC_Order` here, not `WC_Stripe_Order`. So we should keep the previous code in these functions. 

Similar changes (reverts) should be made for other functions here as well to  avoid breaking plugin compaatibility.

Reply:
Alright, sorry about that! I reverted the changes on the deprecated methods and added extra checks for the object class type on the latest commits.